manifests/blackbox_exporter.pp

```diff
@@ -0,0 +1,166 @@
+# Class: prometheus::blackbox_exporter
+#
+# This module manages prometheus blackbox_exporter
+#
+# Parameters:
+#  [*arch*]
+#  Architecture (amd64 or i386)
+#
+#  [*bin_dir*]
+#  Directory where binaries are located
+#
+#  [*config_file*]
+#  Absolute path to configuration file (blackbox module definitions)
+#
+#  [*download_extension*]
+#  Extension for the release binary archive
+#
+#  [*download_url*]
+#  Complete URL corresponding to the where the release binary archive can be downloaded
+#
+#  [*download_url_base*]
+#  Base URL for the binary archive
+#
+#  [*extra_groups*]
+#  Extra groups to add the binary user to
+#
+#  [*extra_options*]
+#  Extra options added to the startup command
+#
+#  [*group*]
+#  Group under which the binary is running
+#
+#  [*init_style*]
+#  Service startup scripts style (e.g. rc, upstart or systemd)
+#
+#  [*install_method*]
+#  Installation method: url or package (only url is supported currently)
+#
+#  [*manage_group*]
+#  Whether to create a group for or rely on external code for that
+#
+#  [*manage_service*]
+#  Should puppet manage the service? (default true)
+#
+#  [*manage_user*]
+#  Whether to create user or rely on external code for that
+#
+#  [*modules*]
+#  Structured, hash of blackbox module definitions for different probe types, converted 
+#  to yaml for blackbox config file
+#
+#  [*os*]
+#  Operating system (linux is the only one supported)
+#
+#  [*package_ensure*]
+#  If package, then use this for package ensure default 'latest'
+#
+#  [*package_name*]
+#  The binary package name - not available yet
+#
+#  [*restart_on_change*]
+#  Should puppet restart the service on configuration change? (default true)
+#
+#  [*service_enable*]
+#  Whether to enable the service from puppet (default true)
+#
+#  [*service_ensure*]
+#  State ensured for the service (default 'running')
+#
+#  [*service_name*]
+#  Name of the node exporter service (default 'blackbox_exporter')
+#
+#  [*user*]
+#  User which runs the service
+#
+#  [*version*]
+#  The binary release version
+#
+# Example for configuring named blackbox modules via hiera
+# details of the format: https://github.com/prometheus/blackbox_exporter/blob/master/CONFIGURATION.md
+#
+# ---
+# prometheus::blackbox::exporter::modules:
+#   simple_ssl:
+#     prober: http
+#     timeout: 10s
+#     valid_status_codes: 2xx
+#     fail_if_not_ssl: true
+#   easy_tcp:
+#     prober: tcp
+#     preferred_ip_protocol: ip4
+class prometheus::blackbox_exporter (
+  Hash $modules              = {},
+  $arch                      = $::prometheus::params::arch,
```

Comment:
please add datatypes to all of those params. An example is https://github.com/voxpupuli/puppet-prometheus/pull/68/files